### PR TITLE
Add item class methods

### LINKED ISSUE #3: [1pt] Add all methods visible in the diagram.
- Implement methods:
    - can_be_archived?() in the Item class
        - should return true if published_date is older than 10 years
        - otherwise, it should return false
    - move_to_archive() in the Item class
        - should reuse can_be_archived?() method
        - should change the archived property to true if the result of the can_be_archived?() method is true
        - should do nothing if the result of the can_be_archived?() method is false

### PR DESCRIPTION
**Pull request by aimalamiri**
## Added methods to the item class
1. Add can_be_archived method
2. Add move_to_archive method

closes #3

### CODE REVIEW

File: classes/item.rb [hunk 1]
```diff
@@ -13,4 +13,17 @@ def initialize(publish_date)
     @publish_date = publish_date
     @archived = false
   end
+
+  def move_to_archived?
```
Comment (Yazino12) [inline classes/item.rb]: - Method name should be `move_to_archive` NOT  `move_to_archived?`

**Review by Yazino12 — CHANGES_REQUESTED** [review]
### Hi @aimalamiri , 

📝👌 **Your project highlights** 🔥😎 
- Implemented method can_be_archived? correctly ✔

You did an awesome job with this task. Your project is almost complete!  There are some requirements before you can merge it. **PLEASE CHECK THE COMMENTS BELOW 👇🏾👇🏾👇🏾**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/item.rb (hunk 1 above)
Comment (aimalamiri) [reply classes/item.rb]: Thanks for reporting!

I renamed the method.

File: classes/item.rb [hunk 2]
```diff
@@ -13,4 +13,17 @@ def initialize(publish_date)
...
+    @archived = true if can_be_archived?
+  end
+
+  private
+
+  def can_be_archived?
+    now = Date.today
+    difference_in_years = ((now - @publish_date)).to_i
+    should_be_archived = (difference_in_years / 365).to_i
+    should_be_archived >= 10
```
Comment (mwenyoa) [inline classes/item.rb]: please change the logic of the ```should_be_archived``` variable to look like this:
should_be_archived > 10
 as stated in the task link below

[task Link ](https://github.com/aimalamiri/Ruby-Catalog/issues/3)

**Review by mwenyoa — CHANGES_REQUESTED** [review]
### REQUIRED CHANGES:

Great work on implementing the said task:
Please the inline comments for the required changes

File: classes/item.rb (hunk 2 above)
Comment (aimalamiri) [reply classes/item.rb]: Thanks, @mwenyoa for indicating the issue.

I finished this.

**Review by Yazino12 — APPROVED** [review]
### Hi @aimalamiri , 

📝👌 **Your project highlights** 🔥😎 
- Linter is passing ✔
- Fixed the changes the previous reviewer addressed ✔

You did an awesome job with your tasks 🚀
Your project is complete!  There is nothing else to say other than... it's time to merge it 🙌 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/item.rb (hunk 2 above)
Comment (Yazino12) [reply classes/item.rb]: Addressed ✔

**Pull request merged**